app/models/concerns/feed_ad_zone.rb

```diff
@@ -0,0 +1,77 @@
+require "active_support/concern"
+
+module FeedAdZone
+  extend ActiveSupport::Concern
+
+  def add_zone(zone)
+    include_zones << zone unless include_zones.include?(zone)
+  end
+
+  def remove_zone(zone)
+    include_zones.reject! { |z| z == zone }
+  end
+
+  def billboard
+    return unless include_zones.present?
+
+    include_zones.include?("billboard") ? 1 : 0
+  end
+
+  def billboard=(val)
+    self.include_zones ||= []
```

Comment:
And for my 3rd comment in this thread...

Maybe it's better if we deviate this UI from Publish?  Publish doesn't make a ton of sense anyways.  So for the allowed values of `include_zones`:

- When `nil` ... show all 4 boxes checked.  And when the user saves with all 4 checked, set to nil.
- When `[]` ... show all 4 boxes unchecked.
- When `["<name>"]` ... check just those 1, 2, or 3 zones.

That make more sense?  It maintains the existing data as well (it can't be an array with 4 items in it ... it becomes nil instead).